### PR TITLE
[libc][math] Fix signed zeros for powf when underflow happens.

### PR DESCRIPTION
**Pull request by lntue**

### CODE REVIEW

File: libc/src/math/generic/powf.cpp [hunk 1]
```diff
@@ -855,9 +855,11 @@ LLVM_LIBC_FUNCTION(float, powf, (float x, float y)) {
...
   exp2_hi_mid_dd.hi = exp2_hi_mid;
 
-  return static_cast<float>(
-             powf_double_double(idx_x, dx, y6, lo6_hi, exp2_hi_mid_dd)) +
-         0.0f;
+  double r_dd = powf_double_double(idx_x, dx, y6, lo6_hi, exp2_hi_mid_dd);
+  float r_f = static_cast<float>(r_dd);
+
+  // Only fix signed zeros for exact zeros results.
+  return r_dd != 0 ? r_f : r_f + 0.0f;
```
Comment (nickdesaulniers) [inline libc/src/math/generic/powf.cpp:862]: Forgive my ignorance, but what does adding `0.0f` to a `float` do?

Comment (lntue) [reply libc/src/math/generic/powf.cpp:862]: it was used to fix signed zeros in some of the exact cases for some rounding modes other than the default one (I can't remember which ones at the moment)

Comment (lntue) [reply libc/src/math/generic/powf.cpp:862]: yes, for rounding modes other than FE_DOWNWARD

**Review by SchrodingerZhu — APPROVED** [review]
LGTM

File: libc/src/math/generic/powf.cpp (hunk 1 above)
Comment (lntue) [reply libc/src/math/generic/powf.cpp]: I actually don't remember the exact cases that required this, and non of the current tests break without this signed zero correction.  So I remove this to simplify things for now.

**Pull request merged**